adafruit_ble_midi.py

```diff
@@ -41,7 +46,7 @@ def __init__(self):
             fixed_length=False,
         )
 
-    def bind(self, service):
+    def bind(self, service: Service) -> _bleio.PacketBuffer:
```

Comment:
This should specifically be `"MIDIService"` since it should really only go with `MIDIService` which is defined later in the file

Reply:
had to add a pylint directive to avoid E0601: Using variable 'MIDIService' before assignment (used-before-assignment), but now it's past pylint, but is blowing up in the build docs workflow?  It's the same error concept, MidiService isn't known at the time it gets to the class def for class _MidiCharacteristic(ComplexCharacteristic).  They reference each other so moving MidiCharacteristic below MIDIService will just move what order the errors occur in.

Any ideas?

Reply:
This is called a forward reference, where you need to type annotated with a variable that hasn't been defined by the time the interpreter will have reached it.  Technically, an annotation can be anything, so most type checkers can understand (and parse) strings of types and resolve them later.  That's why the quotes around `"MIDIService"` are critical.

See how I recently fixed this type annotation, it's also a forward reference.  Using quotes should be all you need, no need to disable any warnings for pylint.

If [PEP 563](https://peps.python.org/pep-0563/) is ever put into action, forward references won't be a thing because type won't be evaluated as they are encountered.  But, for now, quotes are needed.

Reply:
sorry I missed that in your original comment